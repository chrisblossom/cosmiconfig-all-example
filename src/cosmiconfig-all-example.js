import path from 'path';
import cosmiconfig from 'cosmiconfig';

const explorer = cosmiconfig('example');

async function cosmiconfigAllExample(searchPath) {
    const results = [];
    async function getNext(currentPath) {
        const currentResult = await explorer.search(currentPath);

        // if no result found, end search
        if (!currentResult) {
            return;
        }

        // add current result to end of array
        results.push(currentResult);

        const parsedFilePath = path.parse(currentResult.filepath);

        // Get the next parent directory
        const nextPath = path.resolve(parsedFilePath.dir, '../');

        // eslint-disable-next-line consistent-return
        return getNext(nextPath);
    }

    await getNext(searchPath);

    return results;
}

function cosmiconfigAllExampleSync(searchPath) {
    const results = [];
    function getNext(currentPath) {
        const currentResult = explorer.searchSync(currentPath);

        // if no result found, end search
        if (!currentResult) {
            return;
        }

        // add current result to end of array
        results.push(currentResult);

        const parsedFilePath = path.parse(currentResult.filepath);

        // Get the next parent directory
        const nextPath = path.resolve(parsedFilePath.dir, '../');

        // eslint-disable-next-line consistent-return
        return getNext(nextPath);
    }

    getNext(searchPath);

    return results;
}

export { cosmiconfigAllExample, cosmiconfigAllExampleSync };
