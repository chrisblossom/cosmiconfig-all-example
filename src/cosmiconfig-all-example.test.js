import path from 'path';
import { cosmiconfigAllExample } from './cosmiconfig-all-example';

test('finds all configs', async () => {
    const searchPath = path.resolve(__dirname, '__sandbox__/a/b/c/d/e/');
    const result = await cosmiconfigAllExample(searchPath);

    // eslint-disable-next-line no-console
    console.log('result:', result);

    expect(result).toMatchSnapshot();
});
