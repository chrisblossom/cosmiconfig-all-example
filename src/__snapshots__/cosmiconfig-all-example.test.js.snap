// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`finds all configs 1`] = `
Array [
  Object {
    "config": Object {
      "example": "e",
    },
    "filepath": "<PROJECT_ROOT>/src/__sandbox__/a/b/c/d/e/example.config.js",
  },
  Object {
    "config": Object {
      "example": "d",
    },
    "filepath": "<PROJECT_ROOT>/src/__sandbox__/a/b/c/d/example.config.js",
  },
  Object {
    "config": Object {
      "example": "b",
    },
    "filepath": "<PROJECT_ROOT>/src/__sandbox__/a/b/example.config.js",
  },
  Object {
    "config": Object {
      "example": "a",
    },
    "filepath": "<PROJECT_ROOT>/src/__sandbox__/a/example.config.js",
  },
]
`;
